autogen.sh

```diff
@@ -309,67 +309,53 @@ for arg in "$@" ; do
             esac
             ;;
 
-	-do-*)
-	    opt=`echo A$arg | sed -e 's/^A-do-//'`
-	    case $opt in 
-		build-configure|configure) opt=build_configure ;;
-	    esac
-	    var=do_$opt
-
-	    # Check that this opt is known
-	    eval oldval=\$"$var"
-	    if [ -z "$oldval" ] ; then
-		echo "-do-$opt is unrecognized"
-		exit 1
-	    else
-		var=do_$opt
-		eval $var=yes
-	    fi
-	    ;;
-
         -verbose-autoreconf|--verbose-autoreconf)
             autoreconf_args="-vif"
             export autoreconf_args
             ;;
 
-	-with-errmsgs|--with-errmsgs)
-	    do_geterrmsgs=yes
+	-with-autotools=*|--with-autotools=*)
+	    autotoolsdir=`echo "A$arg" | sed -e 's/.*=//'`
```

Comment:
This patch is unrelated to the goal of this PR.

Reply:
There are `do_xxx` that runs a particular step (skipping all other steps), and there are `with_xxx` to turn on/off options. The goal of this PR is to optimistically skip steps and thus need add many `with{out}_xxx` options to fine-tune. The old code kinda mixes `with` and `do` for both, thus I find it is important to refactor and keep the semantics clear. How about I split and do the refactor first?